lib/cli.ex

```diff
@@ -6,14 +6,22 @@ defmodule Onigumo.CLI do
   def main(argv) do
     parsed = OptionParser.parse(argv, aliases: [C: :working_dir], strict: [working_dir: :string])
 
-    with {switches, [component], []} <- parsed,
-         {:ok, module} <- Map.fetch(@components, String.to_atom(component)) do
-      working_dir = Keyword.get(switches, :working_dir, File.cwd!())
-      module.main(working_dir)
-    else
-      {_, _, [_ | _]} -> usage_message()
-      {_, argv, _} when length(argv) != 1 -> usage_message()
-      :error -> usage_message()
+    case parsed do
+      {switches, [component], []} ->
+        case Map.fetch(@components, String.to_atom(component)) do
+          {:ok, module} ->
+            working_dir = Keyword.get(switches, :working_dir, File.cwd!())
+            module.main(working_dir)
+
+          :error ->
+            usage_message()
+        end
+
+      {_, _, [_ | _]} ->
+        usage_message()
+
+      {_, argv, _} when length(argv) != 1 ->
```

Comment:
I think this would work too. `[_ | [_ | _]` is however less straightforward than `[_ | _]`. Moreover, two conditions, one for an empty list, and one for a list with more than one item, would be necessary. 

```suggestion
      {_, [], _} ->
        usage_mesage()

      {_, [_ | [_ | _]], _} ->
        usage_mesage()
```